### PR TITLE
sbt-freestyle cross sbt versioned

### PR DESCRIPTION
**Pull request by juanpedromoreno**
Plugin is now cross-versioned to sbt 1.0.1 and 0.13.16.

### CODE REVIEW

**Review by fedefernandez — APPROVED** [review]
👍

File: src/main/scala-sbt-1.0/compat.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import sbt.io.CopyOptions
+
+object compat {
+
+  lazy val packageDocSettings: Seq[Def.Setting[_]] = Seq(
+    packageDoc in Compile := {
+      val sourceFile = (baseDirectory in LocalRootProject).value / "README.md"
+      val targetFile = crossTarget.value / s"${name.value}-javadoc.jar"
+
+      IO.copy(Seq(sourceFile -> targetFile), CopyOptions().withOverwrite(true)).toSeq
```
Comment (dwijnand) [inline src/main/scala-sbt-1.0/compat.scala:31]: The 0.13 method is still available in sbt 1: https://github.com/sbt/io/blob/v1.0.0/io/src/main/scala/sbt/io/IO.scala#L587-L589

Comment (juanpedromoreno) [reply src/main/scala-sbt-1.0/compat.scala:31]: `CopyOptions` is also available in any other location?

Comment (dwijnand) [reply src/main/scala-sbt-1.0/compat.scala:31]: No CopyOptions is sbt 1 only. What I meant is you could use

```scala
IO.copy(Seq(sourceFile -> targetFile), overwrite = true).toSeq
```

in both sbt 0.13 and sbt 1 and avoid the extra source code files/directories.

Comment (juanpedromoreno) [reply src/main/scala-sbt-1.0/compat.scala:31]: Got it.

However, now I recall why I had to split them up into different pieces of code. `IO.copy` method signature are not compatible among sbt `0.13` and `1.x`.

* 0.13 (3 arguments):
https://github.com/sbt/sbt/blob/0.13/util/io/src/main/scala/sbt/IO.scala#L616
* 1.0 (4 arguments): 
https://github.com/sbt/io/blob/v1.0.0/io/src/main/scala/sbt/io/IO.scala#L587

Am I missing something?

Comment (dwijnand) [reply src/main/scala-sbt-1.0/compat.scala:31]: You're right. Sorry for the noise.

Comment (juanpedromoreno) [reply src/main/scala-sbt-1.0/compat.scala:31]: No problem, thanks anyway :)

**Pull request merged**